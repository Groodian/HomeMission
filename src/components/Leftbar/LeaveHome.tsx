import React from 'react';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  ListItem,
  ListItemIcon,
  ListItemText,
} from '@mui/material';
import { Logout } from '@mui/icons-material';
import { LoadingButton } from '@mui/lab';
import { useTranslation } from 'next-i18next';
import { useSnackbar } from 'notistack';
import {
  HomeDocument,
  useLeaveHomeMutation,
} from '../../lib/graphql/operations/home.graphql';

const LeaveHome: React.FC = () => {
  const { t } = useTranslation('common', { keyPrefix: 'Leftbar.LeaveHome' });
  const { enqueueSnackbar } = useSnackbar();
  const [leaveHome, { loading }] = useLeaveHomeMutation();
  const [open, setOpen] = React.useState(false);

  return (
    <>
      <ListItem button onClick={() => setOpen(true)}>
        <ListItemIcon>
          <Logout />
        </ListItemIcon>
        <ListItemText primary={t('leave-home')} />
      </ListItem>
      <Dialog
        open={open}
        onClose={() => setOpen(false)}
        aria-labelledby="dialog-title"
        maxWidth="md"
      >
        <DialogTitle id="dialog-title">{t('dialog-title')}</DialogTitle>
        <DialogContent>
          <DialogContentText color={(theme) => theme.palette.warning.main}>
            {t('dialog-warning')}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={() => setOpen(false)} disabled={loading}>
            {t('dialog-cancel')}
          </Button>
          <LoadingButton
            loading={loading}
            color="warning"
            onClick={handleLeave}
          >
            {t('leave-home')}
          </LoadingButton>
        </DialogActions>
      </Dialog>
    </>
  );

  async function handleLeave() {
    try {
      await leaveHome({
        update(cache, { data }) {
          if (!data) return;
          cache.writeQuery({
            query: HomeDocument,
            data: { home: data.leaveHome },
          });
        },
      });
      enqueueSnackbar(t('leave-success'), { variant: 'success' });
    } catch (err) {
      enqueueSnackbar(t('leave-error'), { variant: 'error' });
    }
  }
};

export default LeaveHome;
