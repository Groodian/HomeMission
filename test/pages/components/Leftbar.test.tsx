import React from 'react';
import { gql, InMemoryCache } from '@apollo/client';
import { fireEvent, screen } from '@testing-library/react';
import singletonRouter from 'next/router';
import Leftbar from '../../../src/components/Leftbar/Leftbar';
import { render } from '../testUtils';

jest.mock('next/router', () => require('next-router-mock'));

const apolloCache = new InMemoryCache();
apolloCache.writeQuery({
  query: gql`
    query Home {
      home {
        id
        name
        code
        users {
          id
          name
          picture
        }
      }
    }
  `,
  data: {
    home: {
      __typename: 'Home',
      id: '1',
      name: 'Home',
      code: 'ABCDEF',
      users: [
        {
          __typename: 'User',
          id: '1',
          name: 'Name',
          picture: 'Picture',
        },
      ],
    },
  },
});

describe('Leftbar', () => {
  it('matches snapshot', async () => {
    const { asFragment } = await render(<Leftbar />, { apolloCache });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot DE', async () => {
    const { asFragment } = await render(<Leftbar />, {
      language: 'de',
      apolloCache,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('links to overview page', async () => {
    const { getByText } = await render(<Leftbar />, { apolloCache });
    fireEvent.click(getByText('Overview'));
    expect(singletonRouter).toMatchObject({ asPath: 'overview' });
  });

  it('links to statistics page', async () => {
    const { getByText } = await render(<Leftbar />, { apolloCache });
    fireEvent.click(getByText('Statistics'));
    expect(singletonRouter).toMatchObject({ asPath: 'statistics' });
  });

  it('links to history page', async () => {
    const { getByText } = await render(<Leftbar />, { apolloCache });
    fireEvent.click(getByText('Activity history'));
    expect(singletonRouter).toMatchObject({ asPath: 'history' });
  });

  it('shows invite dialog', async () => {
    const { getByText } = await render(<Leftbar />, { apolloCache });
    fireEvent.click(getByText('Invite others'));
    expect(screen.getByText('Invite roommates')).toBeTruthy();
  });

  it('shows leave home dialog', async () => {
    const { getByText } = await render(<Leftbar />, { apolloCache });
    fireEvent.click(getByText('Leave home'));
    expect(
      screen.getByText('Are you sure you want to leave this home?')
    ).toBeTruthy();
  });
});
