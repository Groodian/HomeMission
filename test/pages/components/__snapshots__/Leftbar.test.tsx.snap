// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leftbar matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  z-index: 1199;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  white-space: nowrap;
  width: calc(60px + 1px);
  -webkit-transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  overflow-x: hidden;
}

.emotion-0 .MuiDrawer-paper {
  height: calc(100% - 90px);
  top: 90px;
  box-sizing: border-box;
}

@media (min-width:600px) {
  .emotion-0 {
    width: calc(60px + 1px);
  }
}

.emotion-0 .MuiDrawer-paper {
  width: calc(60px + 1px);
  -webkit-transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  overflow-x: hidden;
}

@media (min-width:600px) {
  .emotion-0 .MuiDrawer-paper {
    width: calc(60px + 1px);
  }
}

.emotion-1 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: none;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  z-index: 1200;
  -webkit-overflow-scrolling: touch;
  position: fixed;
  top: 0;
  outline: 0;
  left: 0;
  border-right: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-3 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-4.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-4.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-4.Mui-disabled {
  opacity: 0.38;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-4.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-5 {
  min-width: 56px;
  color: rgba(0, 0, 0, 0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-7 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: block;
}

.emotion-9 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-22 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
}

.emotion-25 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<div
    class="MuiDrawer-root MuiDrawer-docked emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <ul
          class="MuiList-root MuiList-padding emotion-3"
        >
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                data-testid="DateRangeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Overview
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                data-testid="BarChartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 9.2h3V19H5zM10.6 5h2.8v14h-2.8zm5.6 8H19v6h-2.8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Statistics
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                data-testid="HistoryIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Activity history
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth emotion-22"
          />
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-25"
                data-testid="GroupAddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9V7h-2v2h-2v2h2v2h2v-2h2V9zM8 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 1c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4zm4.51-8.95C13.43 5.11 14 6.49 14 8s-.57 2.89-1.49 3.95C14.47 11.7 16 10.04 16 8s-1.53-3.7-3.49-3.95zm4.02 9.78C17.42 14.66 18 15.7 18 17v3h2v-3c0-1.45-1.59-2.51-3.47-3.17z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Invite others
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-25"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Leave home
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leftbar matches snapshot DE 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  z-index: 1199;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  white-space: nowrap;
  width: calc(60px + 1px);
  -webkit-transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  overflow-x: hidden;
}

.emotion-0 .MuiDrawer-paper {
  height: calc(100% - 90px);
  top: 90px;
  box-sizing: border-box;
}

@media (min-width:600px) {
  .emotion-0 {
    width: calc(60px + 1px);
  }
}

.emotion-0 .MuiDrawer-paper {
  width: calc(60px + 1px);
  -webkit-transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  transition: width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;
  overflow-x: hidden;
}

@media (min-width:600px) {
  .emotion-0 .MuiDrawer-paper {
    width: calc(60px + 1px);
  }
}

.emotion-1 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: none;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  z-index: 1200;
  -webkit-overflow-scrolling: touch;
  position: fixed;
  top: 0;
  outline: 0;
  left: 0;
  border-right: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-3 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-4.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-4.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-4.Mui-disabled {
  opacity: 0.38;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-4.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-5 {
  min-width: 56px;
  color: rgba(0, 0, 0, 0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-7 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: block;
}

.emotion-9 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-22 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
}

.emotion-25 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<div
    class="MuiDrawer-root MuiDrawer-docked emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <ul
          class="MuiList-root MuiList-padding emotion-3"
        >
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                data-testid="DateRangeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Übersicht
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                data-testid="BarChartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 9.2h3V19H5zM10.6 5h2.8v14h-2.8zm5.6 8H19v6h-2.8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Statistiken
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                data-testid="HistoryIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Tätigkeitsverlauf
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth emotion-22"
          />
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-25"
                data-testid="GroupAddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9V7h-2v2h-2v2h2v2h2v-2h2V9zM8 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 1c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4zm4.51-8.95C13.43 5.11 14 6.49 14 8s-.57 2.89-1.49 3.95C14.47 11.7 16 10.04 16 8s-1.53-3.7-3.49-3.95zm4.02 9.78C17.42 14.66 18 15.7 18 17v3h2v-3c0-1.45-1.59-2.51-3.47-3.17z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                Einladen
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button emotion-4"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-25"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                WG verlassen
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
